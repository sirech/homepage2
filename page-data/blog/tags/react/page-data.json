{
    "componentChunkName": "component---src-templates-blog-index-js",
    "path": "/blog/tags/react",
    "result": {"pageContext":{"siteMetadata":{"title":"Mario Fernandez","description":"This is the blog from Mario Fernandez, a Software Developer working for ThoughtWorks","author":"Mario Fernandez","url":"https://hceris.com","twitter":"@sirech"},"group":[{"frontmatter":{"layout":"post","title":"Applications Instead of Libraries","path":"/applications-instead-of-libraries/","categories":["Architecture","React","Micro frontends","JavaScript"],"date":"2021/11/20","draft":false,"description":"Using module federation to implement micro frontends"}},{"frontmatter":{"layout":"post","title":"From Redux to the Context API: A Practical Migration Guide","path":"/redux-to-context-practical-migration-guide/","categories":["React","Redux","Context API","Architecture","State Management"],"date":"2021/06/14","draft":false,"description":"Let's migrate a React application built with Redux to the Context API and see how it looks like"}},{"frontmatter":{"layout":"post","title":"Starting you new React application with Redux? Consider a simpler approach using the Context API","path":"/starting-a-new-app-with-redux-consider-context-api-first/","categories":["React","Redux","Context API","Architecture"],"date":"2021/05/11","draft":false,"description":"Traditionally, starting a new React application means adding Redux for state management. Let's talk about the Context API as an alternative"}},{"frontmatter":{"layout":"post","title":"Authorization Code Flow with PKCE (OAuth) in a React application","path":"/oauth-authorization-code-flow-pkce-for-react/","categories":["Auth0","OAuth","React"],"date":"2021/03/29","draft":false,"description":"SPAs implementing OAuth should pick Authorization Code Flow with PKCE for maximum security. Let's implement it in React with help of Auth0"}},{"frontmatter":{"layout":"post","title":"A directory structure for React projects","path":"/a-directory-structure-for-react-projects/","categories":["React","TypeScript","Code Organization"],"date":"2020/07/19","draft":false,"description":"How do you structure your components in a React application? I like to group mine around features close to the domain enforced by convention"}},{"frontmatter":{"layout":"post","title":"Fail a test in Jest if an unexpected network request happens","path":"/jest-fail-test-if-unexpected-network-request-happens/","categories":["React","Jest","Testing","react-testing-library","create-react-app","axios"],"date":"2019/12/08","draft":false,"description":"Automatically failing unit tests with Jest in React when network requests are made will make your tests more reliable and easier to maintain"}},{"frontmatter":{"layout":"post","title":"Custom Components in Formik","path":"/custom-components-in-formik/","categories":["React","Formik","react-hooks"],"date":"2019/08/15","draft":false,"description":"Formik is the new cool kid on the block for forms in react. Here is how to build custom input elements on top"}},{"frontmatter":{"layout":"post","title":"Migrating to Gatsby 2","path":"/migrating-to-gatsby2/","categories":["React","Gatsby"],"date":"2018/10/03","draft":false,"description":"Gatsby 2 is out! How did the migration go?"}},{"frontmatter":{"layout":"post","title":"Angular from the perspective of a React fan - Part 3","path":"/angular-from-react-part3/","categories":["JavaScript","React","Angular","TypeScript","angular-cli","redux"],"date":"2018/07/07","draft":false,"description":"Third part of my series on Angular, coming from a React background. This article focus on state management"}},{"frontmatter":{"layout":"post","title":"Angular from the perspective of a React fan - Part 2","path":"/angular-from-react-part2/","categories":["JavaScript","React","Angular","TypeScript","angular-cli"],"date":"2018/04/29","draft":false,"description":"Second part of my series on Angular, coming from a React background. This article focus on testing"}},{"frontmatter":{"layout":"post","title":"Angular from the perspective of a React fan - Part 1","path":"/angular-from-react-part1/","categories":["JavaScript","React","Angular","TypeScript","angular-cli"],"date":"2018/03/25","draft":false,"description":"First part of my series on Angular, coming from a React background. This article is about bootstrapping and TypeScript"}},{"frontmatter":{"layout":"post","title":"Migrating to Gatsby","path":"/migrating-to-gatsby/","categories":["React","Gatsby"],"date":"2018/03/18","draft":false,"description":"It all starts here. After using Create React App for a while, I decided to switch to Gatsby for my personal website"}}],"pathPrefix":"blog/tags/react","first":true,"last":true,"index":1,"pageCount":1,"additionalContext":{"siteMetadata":{"title":"Mario Fernandez","description":"This is the blog from Mario Fernandez, a Software Developer working for ThoughtWorks","author":"Mario Fernandez","url":"https://hceris.com","twitter":"@sirech"}}}},
    "staticQueryHashes": []}