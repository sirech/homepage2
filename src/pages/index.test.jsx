import React from 'react'
import { render } from '@testing-library/react'
import Profile from './index'

describe('components', () => {
  describe('Profile', () => {
    it('renders correctly', () => {
      const { asFragment } = render(<Profile />)
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
