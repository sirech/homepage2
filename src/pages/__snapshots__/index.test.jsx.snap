// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Profile renders correctly 1`] = `
<DocumentFragment>
  <div>
    <main>
      <section
        class="horizontal-section"
      >
        <div
          class="container"
        >
          <p>
            I develop software for a living, and then I go home and continue thinking about software because I just can't get enough.
          </p>
          <p>
            I'm a full-stack engineer with infrastructure skills. I've led multiple agile delivery teams, being an individual contributor, driving architecture topics, and coaching and supporting other team members.
          </p>
          <p>
            I believe in high-quality software and advocate for Continuous Delivery, Test Driven Development, and quick iteration. I 
            <a
              href="/blog"
            >
              write
            </a>
             and 
            <a
              href="/talks"
            >
              speak
            </a>
             about my experience regularly. Needless to say, all the opinions expressed in this blog are mine and only mine.
          </p>
        </div>
      </section>
      <section
        class="horizontal-section bg-primary text-white"
        id="techs"
      >
        <div
          class="technologies"
        >
          <div
            class="container"
          >
            <h2
              class="section-heading text-center"
            >
              Technologies
            </h2>
            <p>
              Following the 
              <a
                class="text-dark"
                href="https://www.amazon.com/Pragmatic-Programmer-Journeyman-Master/dp/020161622X"
              >
                Pragmatic Programmer
              </a>
               advice, I try to learn some new technology every year. My github account is full of small projects I have started to teach myself new things, such as this page itself. A list of them, in no particular order
            </p>
            <ul
              class="techList"
            >
              <li
                class="tech"
                data-placement="top"
                data-toggle="tooltip"
                title="Ruby"
              >
                <i
                  class="tech-icon devicon-ruby-plain"
                  data-emergence="hidden"
                />
              </li>
              <li
                class="tech"
                data-placement="top"
                data-toggle="tooltip"
                title="Rails"
              >
                <i
                  class="tech-icon devicon-rails-plain"
                  data-emergence="hidden"
                />
              </li>
              <li
                class="tech"
                data-placement="top"
                data-toggle="tooltip"
                title="JavaScript"
              >
                <i
                  class="tech-icon devicon-javascript-plain"
                  data-emergence="hidden"
                />
              </li>
              <li
                class="tech"
                data-placement="top"
                data-toggle="tooltip"
                title="TypeScript"
              >
                <i
                  class="tech-icon devicon-typescript-plain"
                  data-emergence="hidden"
                />
              </li>
              <li
                class="tech"
                data-placement="top"
                data-toggle="tooltip"
                title="React"
              >
                <i
                  class="tech-icon devicon-react-plain"
                  data-emergence="hidden"
                />
              </li>
              <li
                class="tech"
                data-placement="top"
                data-toggle="tooltip"
                title="SQL"
              >
                <i
                  class="tech-icon devicon-mysql-plain"
                  data-emergence="hidden"
                />
              </li>
              <li
                class="tech"
                data-placement="top"
                data-toggle="tooltip"
                title="Docker"
              >
                <i
                  class="tech-icon devicon-docker-plain"
                  data-emergence="hidden"
                />
              </li>
              <li
                class="tech"
                data-placement="top"
                data-toggle="tooltip"
                title="HTML"
              >
                <i
                  class="tech-icon devicon-html5-plain"
                  data-emergence="hidden"
                />
              </li>
              <li
                class="tech"
                data-placement="top"
                data-toggle="tooltip"
                title="CSS"
              >
                <i
                  class="tech-icon devicon-css3-plain"
                  data-emergence="hidden"
                />
              </li>
              <li
                class="tech"
                data-placement="top"
                data-toggle="tooltip"
                title="Python"
              >
                <i
                  class="tech-icon devicon-python-plain"
                  data-emergence="hidden"
                />
              </li>
              <li
                class="tech"
                data-placement="top"
                data-toggle="tooltip"
                title="Java"
              >
                <i
                  class="tech-icon devicon-java-plain"
                  data-emergence="hidden"
                />
              </li>
              <li
                class="tech"
                data-placement="top"
                data-toggle="tooltip"
                title="AWS"
              >
                <i
                  class="tech-icon devicon-amazonwebservices-plain"
                  data-emergence="hidden"
                />
              </li>
              <li
                class="tech"
                data-placement="top"
                data-toggle="tooltip"
                title="Kubernetes"
              >
                <i
                  class="tech-icon devicon-kubernetes-plain"
                  data-emergence="hidden"
                />
              </li>
            </ul>
          </div>
        </div>
      </section>
      <section
        class="horizontal-section"
        id="tools"
      >
        <div
          class="container"
        >
          <h2
            class="section-heading text-center"
          >
            Tools
          </h2>
          <hr
            class="border-white"
          />
          <p>
            I am a bit picky about the tools I use. Actually, I am 
            <b>
              very picky
            </b>
            . I cannot remember how many times I did rewrite my shell config in the past. It is problematic for pair programming because I usually am the only one that can work at my computer.
          </p>
          <p>
            Nowadays I try to at least outsource my configurations a bit, using some awesome community driven ones like 
            <a
              href="http://spacemacs.org"
              rel="noopener noreferrer"
              target="_blank"
            >
              Spacemacs
            </a>
             and 
            <a
              href="https://github.com/robbyrussell/oh-my-zsh"
              rel="noopener noreferrer"
              target="_blank"
            >
              Oh My Zsh
            </a>
            .
          </p>
          <ul
            class="text-center"
          >
            <li
              class="me-1 mb-1 badge bg-secondary rounded-pill"
            >
              spacemacs
            </li>
            <li
              class="me-1 mb-1 badge bg-secondary rounded-pill"
            >
              zsh
            </li>
            <li
              class="me-1 mb-1 badge bg-secondary rounded-pill"
            >
              git
            </li>
            <li
              class="me-1 mb-1 badge bg-secondary rounded-pill"
            >
              tmux
            </li>
            <li
              class="me-1 mb-1 badge bg-secondary rounded-pill"
            >
              Google Chrome
            </li>
            <li
              class="me-1 mb-1 badge bg-secondary rounded-pill"
            >
              hammerspoon
            </li>
            <li
              class="me-1 mb-1 badge bg-secondary rounded-pill"
            >
              Alfred
            </li>
            <li
              class="me-1 mb-1 badge bg-secondary rounded-pill"
            >
              Dash
            </li>
            <li
              class="me-1 mb-1 badge bg-secondary rounded-pill"
            >
              trello
            </li>
          </ul>
        </div>
      </section>
    </main>
    <footer
      class="footer text-center"
    >
      <div>
        Copyright Â© 2022, 
        <a
          href="http://creativecommons.org/licenses/by-nc-sa/4.0/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Licensed under Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License.
        </a>
      </div>
      Favicon courtesy of 
      <a
        href="https://icons8.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Icons8
      </a>
       
    </footer>
  </div>
</DocumentFragment>
`;
